addons/mrp_subcontracting_account/models/stock_move.py

```diff
@@ -51,6 +51,10 @@ def _generate_valuation_lines_data(self, partner_id, qty, debit_value, credit_va
                 'balance': -component_cost,
                 'account_id': credit_account_id,
             }
+        # if svl passed is not linked to the move in self, the valuation is a correction and should always credit the
+        # `stock_input` account as it adds directly to the value of the subcontracted product
+        if svl_id and self.stock_valuation_layer_ids.ids and svl_id not in self.stock_valuation_layer_ids.ids:
+            rslt['credit_line_vals']['account_id'] = self.product_id.product_tmpl_id.get_product_accounts()['stock_input'].id
```

Comment:
I don't find this very safe to write on `['credit_line_vals']` while the precedent `if` may erase it 
https://github.com/odoo/odoo/blob/2926757ff67087d172bdcfdeccb133b4024b4081/addons/mrp_subcontracting_account/models/stock_move.py#L32. Even tho I tried to have a rtraceback and I dont think this is feasible to go throuth the two `ifs` in a same function call. 
I would anyway move your patch above

Reply:
You're right, i didn't notice it because both ifs cannot run in one call, I can't remember why tho xD